GitExtUtils/FileLogging/AsynchronousLogWriter.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace GitExtUtils.FileLogging
+{
+    public class AsynchronousLogWriter
+    {
+        private readonly LogFilenameCreator _logFilenameCreator;
+        private readonly ILogFormatter _formatter;
+        private readonly TimeSpan _flushInterval;
+        private readonly CancellationToken _cancellationToken;
+        private readonly ConcurrentQueue<LogEntry> _unsaved = new ConcurrentQueue<LogEntry>();
+
+        public AsynchronousLogWriter(
+            LogFilenameCreator logFilenameCreator,
+            ILogFormatter formatter,
+            TimeSpan flushInterval,
+            CancellationToken cancellationToken)
+        {
+            _logFilenameCreator = logFilenameCreator;
+            _formatter = formatter;
+            _flushInterval = flushInterval;
+            _cancellationToken = cancellationToken;
+        }
+
+        public void Enqueue(LogEntry entry)
+        {
+            _unsaved.Enqueue(entry);
+        }
+
+        public void Run()
```

Comment:
`CancellationToken` looks very odd as a property of the writer, it is a part that is directly related to the execution, e.g. _run until cancelled_.
So I think it would be more appropriate to do the following:
```suggestion
        public void Run(CancellationToken cancellationToken)
```

Reply:
agree